osu.Game/Rulesets/Mods/ModFlashlight.cs

```diff
@@ -123,16 +122,22 @@ protected override void LoadComplete()
             {
                 base.LoadComplete();
 
-                Combo.ValueChanged += _ => UpdateFlashlightSize(GetSize());
+                FlashlightSize = adjustCurrentSize();
+
+                Combo.ValueChanged += _ => updateFlashlightSize();
 
                 if (player != null)
                 {
                     isBreakTime.BindTo(player.IsBreakTime);
-                    isBreakTime.BindValueChanged(_ => UpdateFlashlightSize(GetSize()), true);
+                    isBreakTime.BindValueChanged(_ => updateFlashlightSize(), true);
                 }
             }
 
-            protected abstract void UpdateFlashlightSize(float size);
+            private Vector2 adjustCurrentSize() => AdjustSize(GetSize());
+
+            private void updateFlashlightSize() => this.TransformTo(nameof(FlashlightSize), adjustCurrentSize(), 800);
+
+            protected abstract Vector2 AdjustSize(float size);
```

Comment:
Probably a good opportunity to add some simple xmldoc here explaining the input and output.

Reply:
I believe done?